apps/omg/test/support/conformance/merkle_proof_context.ex

```diff
@@ -0,0 +1,208 @@
+    %{base | leaf: <<0::256>>}
+  end
+
+  defp random_leaf(%__MODULE__{} = base) do
+    let b <- such_that(b <- binary(), when: b != base.leaf) do
+      %{base | leaf: b}
+    end
+  end
+
+  defp trimmed_leaf(%__MODULE__{} = base) do
+    length_leaf = byte_size(base.leaf)
+
+    let to_keep <- integer(0, length_leaf - 1) do
+      %{base | leaf: binary_part(base.leaf, 0, to_keep)}
+    end
+  end
+
+  defp expanded_leaf(%__MODULE__{} = base) do
+    let [b <- non_empty_binary(), append? <- boolean()] do
+      if append?, do: %{base | leaf: base.leaf <> b}, else: %{base | leaf: b <> base.leaf}
+    end
+  end
+
+  defp get_other_leaf(%__MODULE__{} = base) do
+    length_leaves = length(base.leaves)
+    distinct_leaf_index = such_that(i <- integer(0, length_leaves - 1), when: Enum.at(base.leaves, i) != base.leaf)
+
+    let other_index <- distinct_leaf_index do
+      %{base | leaf: Enum.at(base.leaves, other_index)}
+    end
+  end
+
+  #
+  # proof mutations
+
+  defp bitwise_modify_proof(%__MODULE__{} = base) do
+    # TODO: more cases pending
+    union([
+      bitwise_append(base)
+    ])
+  end
+
+  defp chunkwise_modify_proof(%__MODULE__{} = base) do
+    insert_leaf_chunk_weight = if base.leaf == "", do: 0, else: 1
+
+    weighted_union([
+      {1, insert_zero_chunk(base)},
+      {insert_leaf_chunk_weight, insert_leaf_chunk(base)},
+      {1, drop_chunk(base)},
+      {1, swap_neighbors(base)}
+    ])
+  end
+
+  defp insert_zero_chunk(%__MODULE__{} = base) do
+    # @proof length doesn't work for some reason
+    let position <- integer(0, 16) do
+      %{base | proof: base.proof |> chunk() |> List.insert_at(position, <<0::256>>) |> unchunk()}
+    end
+  end
+
+  defp insert_leaf_chunk(%__MODULE__{} = base) do
+    # @proof length doesn't work for some reason
+    let position <- integer(0, 16) do
+      %{base | proof: base.proof |> chunk() |> List.insert_at(position, base.leaf) |> unchunk()}
+    end
+  end
+
+  defp drop_chunk(%__MODULE__{} = base) do
+    # @proof length doesn't work for some reason
+    let position <- integer(0, 16 - 1) do
+      %{base | proof: base.proof |> chunk() |> List.delete_at(position) |> unchunk()}
+    end
+  end
+
+  defp swap_neighbors(%__MODULE__{} = base) do
+    # @proof length doesn't work for some reason
+    let position <- integer(0, 16 - 1 - 1) do
+      chunked_proof = chunk(base.proof)
+      [neighbor1, neighbor2] = Enum.slice(chunked_proof, position, 2)
+
+      swapped_proof =
+        chunked_proof
+        |> List.replace_at(position, neighbor2)
+        |> List.replace_at(position + 1, neighbor1)
+        |> unchunk
+
+      %{base | proof: swapped_proof}
+    end
+  end
+
+  defp bitwise_append(%__MODULE__{} = base) do
+    let to_append <- union([non_empty_binary(), <<0>>, <<0::256>>, <<0::512>>]) do
+      %{base | proof: base.proof <> to_append}
+    end
+  end
+
+  defp non_empty_binary(), do: such_that(b <- binary(), when: b != "")
+
+  #
```

Comment:
Unnecessary line.

Reply:
hm, not sure which line do you mean? this is just an eye-candy header to divide code into sections.

```

#
# header

```

or am I missing sth?